### ISSUE TITLE
all_gather fails on certain tensor shapes

### ISSUE THREAD

**Issue by handrewsTT** [tenstorrent/tt-metal#29518]
### Describe the bug

After a recent change to `tt-metal` main (#27287) we see certain models failing at `ttnn.experimental.all_gather_async` with certain tensor shapes.

When calling `ttnn.experimental.all_gather_async` with certain tensor shapes we get the error:
```                 from /home/ubuntu/actions-runner/_work/tt-metal/tt-metal/tt_metal/hw/firmware/src/tt-1xx/brisck.cc:21:
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/ttnn/cpp/ttnn/operations/experimental/ccl/all_broadcast_async/device/kernels/all_broadcast_rm_writer.cpp: In function 'void kernel_main()':
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/ttnn/cpp/ttnn/operations/experimental/ccl/all_broadcast_async/device/kernels/all_broadcast_rm_writer.cpp:101:71: error: conversion from 'long unsigned int' to 'uint16_t' {aka 'short unsigned int'} changes value from '65568' to '32' [-Werror=overflow]
  101 |         fabric_connection, unicast_route_id, starts, ranges, nullptr, page_size);
      |                                                                       ^~~~~~~~~
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/ttnn/cpp/ttnn/operations/experimental/ccl/all_broadcast_async/device/kernels/all_broadcast_rm_writer.cpp:111:19: error: conversion from 'long unsigned int' to 'uint16_t' {aka 'short unsigned int'} changes value from '131136' to '64' [-Werror=overflow]
  111 |         page_size * 2);
```

### Steps to reproduce the issue

pytest to reproduce below. Fails on (1, 1, 32, 32776) and (1, 1, 32, 18992) but passes on (1, 1, 32, 16032)
```
import ttnn
import torch
from models.tt_transformers.tt.ccl import TT_CCL
import pytest

@pytest.mark.parametrize("device_params", [{"fabric_config": ttnn.FabricConfig.FABRIC_1D}], indirect=True)
@pytest.mark.parametrize("mesh_device", [(1, 8)], indirect=True)
@pytest.mark.parametrize("tensor_shape", [
    (1, 1, 32, 32776),
    (1, 1, 32, 18992),
    (1, 1, 32, 16032),
])
def test_all_gather_async(mesh_device, tensor_shape):

    x = ttnn.from_torch(torch.randn(tensor_shape), layout=ttnn.TILE_LAYOUT, device=mesh_device, mesh_mapper=ttnn.ReplicateTensorToMesh(mesh_device), memory_config=ttnn.DRAM_MEMORY_CONFIG)
    tt_ccl = TT_CCL(mesh_device)
    output = ttnn.experimental.all_gather_async(
        x,
        persistent_output_buffer=None,
        dim=3,
        multi_device_global_semaphore=tt_ccl.get_and_cycle_ag_semaphore_handles(None),
        num_links=1,
        memory_config=x.memory_config(),
        cluster_axis=None,
        topology=ttnn.Topology.Ring,
        barrier_semaphore=tt_ccl.get_and_cycle_barrier_semaphore_handle(None),
        chunks_per_sync=10,
        num_workers_per_link=2,
        num_buffers_per_channel=2,
    )

```

**Comment by handrewsTT** [comment]
@cmaryanTT please could you assign this to the appropriate person on the ops team

**Comment by ejouretTT** [comment]
Notes
- Seems to be an issue on fabric side - failed on CI. 

Next
- @handrewsTT @mtairum try workaround of adding padding to Gemma -> see if hang goes away 
- Need to assign a singular owner @ntarafdar 
- @handrewsTT to add link to conversation relating to origin of issue

**Comment by ntarafdar** [comment]
might be related to this issue: https://github.com/tenstorrent/tt-metal/issues/29349
@nardoTT will verify if its the same, she has a solution for the above issue

**Comment by handrewsTT** [comment]
Issue raised after push back on revert PR https://github.com/tenstorrent/tt-metal/pull/29416

**Comment by nardoTT** [comment]
This should be fixed in this [PR.](https://github.com/tenstorrent/tt-metal/pull/29572) Running this test after is passing:

```
import torch
import pytest
import math
from loguru import logger
import ttnn
from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
from tests.ttnn.unit_tests.operations.ccl.test_all_gather import is_unsupported_case
from models.common.utility_functions import skip_for_blackhole
from tests.nightly.t3000.ccl.test_minimal_all_gather_async import run_all_gather_impl


@pytest.mark.parametrize("num_links", [1], ids=["1link"])
@pytest.mark.parametrize("mesh_device", [(1, 8)], indirect=True)
@pytest.mark.parametrize(
    "ag_output_shape, dim, layout, ag_input_dtype",
    [
        ([1, 1, 32, 32776], 3, ttnn.TILE_LAYOUT, ttnn.bfloat16),
        ([1, 1, 32, 18992], 3, ttnn.TILE_LAYOUT, ttnn.bfloat16),
        ([1, 1, 32, 16032], 3, ttnn.TILE_LAYOUT, ttnn.bfloat16),
    ],
    ids=[
        "test_one",
        "test_two",
        "test_three",
    ],
)
@pytest.mark.parametrize(
    "mem_config_input, mem_config_ag",
    [
        (
            ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM),
            ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM),
        )
    ],
)
@pytest.mark.parametrize(
    "enable_trace, num_iters",
    [
        (True, 10),
        (False, 1),
    ],
    ids=["perf", "check"],
)
@pytest.mark.parametrize(
    "device_params, all_gather_topology",
    [
        ({"fabric_config": ttnn.FabricConfig.FABRIC_1D, "trace_region_size": 103424}, ttnn.Topology.Ring),
    ],
    indirect=["device_params"],
    ids=["fabric_ring"],
)
def test_all_gather_failing_shapes(
    mesh_device,
    ag_output_shape,
    dim,
    num_links,
    ag_input_dtype,
    layout,
    mem_config_input,
    mem_config_ag,
    enable_trace,
    all_gather_topology,
    num_iters,
):
    run_all_gather_impl(
        mesh_device,
        mesh_device.get_num_devices(),
        ag_output_shape,
        dim,
        num_links,
        ag_input_dtype,
        layout,
        mem_config_input,
        mem_config_ag,
        all_gather_topology=all_gather_topology,
        enable_trace=enable_trace,
        num_iters=num_iters,
        use_barrier=True,
        use_persistent_buffers=False,
        chunks_per_sync=10,
        num_workers_per_link=2,
        num_buffers_per_channel=2,
    )
```

**Comment by ejouretTT** [comment]
Change merged.

**Comment by ntarafdar** [comment]
Making a comment here to track root cause, the root cause was that after a fabric uplift this exposed code in the all_broadcast CCL op (used in certain configs of he all-gather) where the packet size was incorrect. This would have been caught if all multi-chip pipelines were run after fabric changes. We will bring this up as best-practices in fabric/CCL team.
